clippy_lints/src/functions/ref_option.rs

```diff
@@ -0,0 +1,108 @@
+use crate::functions::hir::GenericArg;
+use crate::functions::REF_OPTION;
+use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::source::snippet;
+use rustc_errors::Applicability;
+use rustc_hir as hir;
+use rustc_hir::intravisit::FnKind;
+use rustc_hir::{FnDecl, GenericArgs, ImplItem, MutTy, QPath, TraitItem, Ty, TyKind};
+use rustc_lint::LateContext;
+use rustc_span::{sym, Span};
+
+fn check_ty(cx: &LateContext<'_>, param: &Ty<'_>, fixes: &mut Vec<(Span, String)>) {
+    // TODO: is this the right way to check if ty is an Option?
+    if let TyKind::Ref(lifetime, MutTy { ty, .. }) = param.kind
+        && let TyKind::Path(QPath::Resolved(_, path)) = ty.kind
+        && path.segments.len() == 1
+        && let seg = &path.segments[0]
+        && seg.ident.name == sym::Option
```

Comment:
I think I know...but I'd like to share my reasoning here so others reading this can follow: We have other Option-based lints, so grepping for Option should yield something illuminating. Given that I want to match on a `sym` (which you already checked), I `rg sym::Option clippy_lints`, which gives me `is_type_diagnostic_item(cx, ty, sym::Option)`. That should work beautifully in your case and doesn't fail if someone writes `std::option::Option` instead of plain `Option`. In general, grepping the code is a good way to find helpful implementation techniques.